homeassistant/components/mqtt/mixins.py

```diff
@@ -223,6 +224,27 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
 )
 
 
+def validate_modern_schema(domain: str) -> Callable:
```

Comment:
The function name is confusing since this doesn't validate the modern schema, it warns when the old schema is used.